### PR TITLE
Enable json in maven plugin

### PR DESCRIPTION
**Pull request by blacelle**
Following https://github.com/diffplug/spotless/issues/1445

---

Please **DO NOT FORCE PUSH**. Don't worry about messy history, it's easier to do code review if we can tell what happened after the review, and force pushing breaks that.

Please make sure that your [PR allows edits from maintainers](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/).  Sometimes its faster for us to just fix something than it is to describe how to fix it.

Allow edits from maintainers

After creating the PR, please add a commit that adds a bullet-point under the `[Unreleased]` section of [CHANGES.md](https://github.com/diffplug/spotless/blob/main/CHANGES.md), [plugin-gradle/CHANGES.md](https://github.com/diffplug/spotless/blob/main/plugin-gradle/CHANGES.md), and [plugin-maven/CHANGES.md](https://github.com/diffplug/spotless/blob/main/plugin-maven/CHANGES.md) which includes:

- [ ] a summary of the change
- either
    - [ ] a link to the issue you are resolving (for small changes)
    - [ ] a link to the PR you just created (for big changes likely to have discussion)

If your change only affects a build plugin, and not the lib, then you only need to update the `plugin-foo/CHANGES.md` for that plugin.

If your change affects lib in an end-user-visible way (fixing a bug, updating a version) then you need to update `CHANGES.md` for both the lib and all build plugins.  Users of a build plugin shouldn't have to refer to lib to see changes that affect them.

This makes it easier for the maintainers to quickly release your changes :)

### CODE REVIEW

File: plugin-maven/README.md [hunk 1]
```diff
@@ -700,6 +701,53 @@ The auto-discovery of config files (up the file tree) will not work when using t
 
 For details, see the [npm detection](#npm-detection) and [`.npmrc` detection](#npmrc-detection) sections of prettier, which apply also to tsfmt.
 
+## JSON
+
+- `com.diffplug.spotless.maven.json.Json` [javadoc](https://javadoc.io/doc/com.diffplug.spotless/spotless-plugin-gradle/6.12.1/com/diffplug/gradle/spotless/JsonExtension.html), [code](https://github.com/diffplug/spotless/blob/main/plugin-maven/src/main/java/com/diffplug/spotless/maven/json/Json.java)
```
Comment (nedtwigg) [inline plugin-maven/README.md:706]: The javadoc link is to plugin-gradle instead of plugin-maven

**Review by nedtwigg — CHANGES_REQUESTED** [review]
Great PR! Needs

- an entry in `plugin-maven/CHANGES.md`
- at least one test for Gson and one test Simple

**Comment by nedtwigg** [comment]
Oh, also run `spotlessApply` and commit the result ;-)

**Review by nedtwigg — APPROVED** [review]
Great PR, thanks!

**Comment by blacelle** [comment]
@nedtwigg Would it be reasonable to add `'**/*.json' `as default include? (It is unclear to me why some formats has a default while other do not (which I feel there is some rationale default include)).

**Comment by nedtwigg** [comment]
> Would it be reasonable to add '**/*.json' as default include?

We used to have a lot of `**/*.blah` default includes, and it turned out to be a mistake. You have a toy project, the `**/*` works really well. People with real projects actually adopt it, and those wildcards become meaningfully slow.

> why some formats has a default while other do not

There are some cases where a standard is common enough that most users won't need to change it. e.g. `src/main/java/**/*.java`. But unless it's really standard *and contained to a specific folder* then it's better for it to be manually specified.

The great underappreciated value of needing to manually specify the target is that it's self-documenting. Someone new joins the project, is confused by what's getting formatted, searches "spotless" in their project. And right there, they can see how to change it. If the target hasn't been set, the default is an invisible mystery.

Invisible mysteries are nice, but only when they work, so it's important to only set a default if it's almost definitely right. We got this wrong earlier in Spotless' life.

**Comment by nedtwigg** [comment]
Published in `plugin-maven 2.30.0`.

**Pull request merged**